psutil/tests/__init__.py

```diff
@@ -380,20 +375,26 @@ def wait_for_pid(pid, timeout=GLOBAL_TIMEOUT):
             return
 
 
-def wait_for_file(fname, timeout=GLOBAL_TIMEOUT, delete_file=True):
-    """Wait for a file to be written on disk with some content."""
+def wait_for_file(fname, timeout=GLOBAL_TIMEOUT, empty=False,
+                  delete_file=True):
+    """Wait for a file to be written on disk with some content, or until
+    the file exists if empty=True."""
     stop_at = time.time() + timeout
+    sleep_for = 0.001
     while time.time() < stop_at:
         try:
             with open(fname, "r") as f:
                 data = f.read()
-            if not data:
+            if not empty and not data:
+                time.sleep(sleep_for)
+                sleep_for *= 2
                 continue
             if delete_file:
                 os.remove(fname)
             return data
-        except IOError:
-            time.sleep(0.001)
+        except (IOError, OSError):
```

Comment:
This is too broad. We should only `pass` in case of `OSError` + `ENOENT` (meaning file is still not there) and in case of windows also take `WindowsError` + `errno == 32` (file used by another process) into account.
This way we'll get an exception if, e.g., the file cannot be opened or removed because of permission issues (we want to be aware of that).